src/chain/payout.rs

```diff
@@ -101,17 +126,23 @@ pub async fn payout(
                 "{batch_transaction:?}"
             )))?;
 
-        let signature = signer.sign(order.id, sign_this).await?;
+        let signature = signer.sign(order.id.clone(), sign_this).await?;
 
-        batch_transaction.signature.content =
-            TypeContentToFill::SpecialType(SpecialTypeToFill::SignatureSr25519(Some(signature)));
+        if let TypeContentToFill::Variant(ref mut multisig) = batch_transaction.signature.content {
+            if let TypeContentToFill::ArrayU8(ref mut sr25519) =
+                multisig.selected.fields_to_fill[0].type_to_fill.content
+            {
+                sr25519.content = signature.0.to_vec();
+            }
+        }
 
         let extrinsic = batch_transaction
             .send_this_signed::<(), RuntimeMetadataV15>(&chain.metadata)?
             .ok_or(ChainError::NothingToSend)?;
 
         send_stuff(&client, &format!("0x{}", const_hex::encode(extrinsic))).await?;
 
+        state.order_withdrawn(order.id.clone()).await;
```

Comment:
is this clone needed?

Reply:
resolved in [16fe114](https://github.com/Alzymologist/Kalatori-backend/pull/106/commits/16fe114e333990884ec4beabb69f1e5518c74cde)